opscx/opscx_connector.lua

```diff
@@ -0,0 +1,627 @@
+  if e.element == 14 or e.element == 24 then
+    local is_downtime = "false"
+    local is_acknowledged = "false"
+    local is_flapping = "false"
+    local check_type = "active"
+    local output = string.gsub(e.output,"\\n","\n") .. metrics
+    if e.check_type == 1 then
+      check_type = "passive"
+    end
+    if e.is_flapping then
+      is_flapping = "true"
+    end
+    if e.acknowledged then
+      is_acknowledged = "true"
+    end
+    if e.scheduled_downtime_depth and e.scheduled_downtime_depth == 1 then
+      is_downtime = "true"
+    end
+    event_to_send = event_to_send ..
+      "<event_data xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:noNamespaceSchemaLocation=\"../docs/schema/event.xsd\">" ..
+      "<title>" .. service_description .. "</title>" .. 
+      "<description>" .. string.sub(ifnil_or_empty(string.match(output, "^(.*)\n"), 'no output'), 1, self.max_output_length) .. "</description>" .. 
+      "<severity>" .. state .. "</severity>" .. 
+      "<state></state>" ..
+      "<time_created>" .. e.last_state_change .. "</time_created>" ..
+      "<category></category>" ..
+      "<subcategory></subcategory>" ..
+      "<eti></eti>" ..
+      "<node>" .. hostname .. "</node>" .. 
+      "<related_ci></related_ci>" ..
+      "<subcomponent></subcomponent>" ..
+      "<source_ci>" .. self.source_ci .. "</source_ci>" .. 
+      "<source_event_id></source_event_id>" ..
+      "<source_event_url>" .. source_event_url .. "</source_event_url>" .. 
+      "<omi_event_key>" .. omi_event_key .. "</omi_event_key>" ..
+      "<custom_attributes>" ..
+      "<name01>is_acknowledged</name01>" ..
+      "<value01>" .. is_acknowledged .. "</value01>" ..
+      "<name02>is_downtime</name02>" ..
+      "<value02>" .. is_downtime .. "</value02>" ..
+      "<name03></name03>" ..
+      "<value03></value03>" ..
+      "<name04>is_flapping</name04>" ..
+      "<value04>" .. is_flapping .. "</value04>" ..
+      "<name05>check_type</name05>" ..
+      "<value05>" .. check_type .. "</value05>" ..
+      "<name06>time_updated</name06>" ..
+      "<value06>" .. format_date(new_from_timestamp(get_tz():localise(e.last_state_change))) .. "</value06>" ..
+      "<name07>hostgroup</name07>" ..
+      "<value07>" .. hostgroup_to_send .. "</value07>" ..
+      "<name08>host_severity</name08>" ..
+      "<value08>" .. host_severity .. "</value08>" ..
+      "<name09></name09>"..
+      "<value09></value09>" ..
+      "<name10></name10>"..
+      "<value10></value10>" ..
+      "</custom_attributes>" ..
+      "</event_data>"
+  end
+
+  -- Appending to the event queue
+  self.events[#self.events + 1] = event_to_send
+  return true
+
+end
+
+--------------------------------------------------------------------------------
+-- EventQueue:flush method
+-- Called when the max number of events or the max age are reached
+--------------------------------------------------------------------------------
+
+function EventQueue:flush()
```

Comment:
I recommend also to add a function flush(), something like this:

```
function flush()
  return queue:flush()
end
```

This function is really important in case of retention files. Broker does not call anymore the write() function while there are retention files. But it tries to call the flush(), the goal is to empty the retention files.

So in case there is no flush() function, the retention files could increase and the stream connector would be locked.

Reply:
@pkriko if you don't mind ;) then we will be able to merge this one